### PR TITLE
fix(datagrid): update column states on detached from DOM rows

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Detached from DOM `clr-dg-row` doesnt update their cells dimensions.
Discovered in virtual scroller.

Issue Number: CDE-3092

## What is the new behavior?
All`clr-dg-row` now have listener for column state changes that will update their cells dimensions if the row is detached from DOM.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/data/datagrid/datagrid-virtual-scroll.directive.ts [hunk 1]
```diff
@@ -291,6 +291,11 @@ export class ClrDatagridVirtualScrollDirective<T> implements AfterViewInit, DoCh
         if (datagridState.filters) {
           this.scrollToIndex(0);
         }
+      }),
+      this.columnsService.columnsStateChange.subscribe(() => {
+        // Simply, there is no correct way to clear cached rows upon column resize from the view repeater.
+        // _viewRepeater is a private field and cdkVirtualFor doesn't provide a public API for it.
+        (this.cdkVirtualFor as any)._viewRepeater.detach();
```
Comment (dtsanevmw) [inline projects/angular/data/datagrid/datagrid-virtual-scroll.directive.ts]: The API of this says it's doing the same thing, did you try it?
```suggestion
        this.virtualScrollViewport.detach()
```

Comment (valentin-mladenov) [reply projects/angular/data/datagrid/datagrid-virtual-scroll.directive.ts]: `CdkVirtualScrollViewport` detach is different from `CdkVirtualForOf ViewRepeater` detach.

The frirst detaches the `CdkVirtualForOf`, but not it's repeater.
Angular doesn't give easy access tothe cached row views, that have to be recalculated.

By clearing the cache we are forcing `CdkVirtualForOf` to take the new calculated rows.

Comment (valentin-mladenov) [reply projects/angular/data/datagrid/datagrid-virtual-scroll.directive.ts]: The `viewRepeater.detach()` is introduced in #1452 and was removed in `angular 21` migration #2204.
I'm simply reintroducing it.

File: projects/angular/data/datagrid/render/row-renderer.ts [hunk 2]
```diff
@@ -39,6 +52,20 @@ export class DatagridRowRenderer implements AfterContentInit, OnDestroy {
     );
   }
 
+  afterNextRender() {
```
Comment (dtsanevmw) [inline projects/angular/data/datagrid/render/row-renderer.ts]:
```suggestion
  listenForCachedRowsColumnStateChanges() {
```

**Review by dtsanevmw — DISMISSED** [review]

File: projects/angular/data/datagrid/render/row-renderer.ts (hunk 2 above)
Comment (valentin-mladenov) [reply projects/angular/data/datagrid/render/row-renderer.ts]: Used `addDetachedRowsColumnStateListener` name.

**Review by dtsanevmw — APPROVED** [review]
